### PR TITLE
This PR enables pipeline hot-reloading

### PR DESCRIPTION
**Pull request by johannesugb**
Description here in the documentation section: https://github.com/cg-tuwien/VulkanLaunchpad/blob/pipeline-hot-reloading/README.md#pipeline-hot-reloading

### CODE REVIEW

**Comment by johannesugb** [comment]
Ah, if compilation fails, the application crashes. That would be a nice-to-not-have. ^^

**Comment by johannesugb** [comment]
If shader compilation fails, the program crashes no longer durinig hot-reloading (only during user-triggered loading through `vklCreateGraphicsPipeline` (as before)).

File: VulkanLaunchpad.cpp [hunk 1]
```diff
@@ -411,7 +419,8 @@ std::vector<uint32_t> compileShaderSourceToSpirv(const std::string& shaderSource
 			      << "\n         Log[" << glslang_shader_get_info_log(shader) << "]"
 			      << "\n         Debug-Log[" << glslang_shader_get_info_debug_log(shader) << "]" << std::endl;
 
-		VKL_EXIT_WITH_ERROR("glslang_shader_preprocess failed for " + inputFilename);
+		// VKL_EXIT_WITH_ERROR("glslang_shader_preprocess failed for " + inputFilename);
```
Comment (electrocnic) [inline VulkanLaunchpad.cpp]: The comments in these lines are intentional?
Or can they be removed for the merge?

**Comment by electrocnic** [comment]
otherwise, lgtm, but I would wait for a second reviewer :)

File: VulkanLaunchpad.cpp (hunk 1 above)
Comment (johannesugb) [reply VulkanLaunchpad.cpp]: Yeah, they can be removed.

**Pull request merged**